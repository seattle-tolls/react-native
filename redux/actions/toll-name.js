import { SWITCH_TOLL_NAME } from '../constants/action-types'

export const switchTollName = (payload) => ({
  type: SWITCH_TOLL_NAME, payload,
})