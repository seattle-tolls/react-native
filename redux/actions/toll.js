import { SET_TOLL_INFO, FETCH_TOLL_INFO } from '../constants/action-types'

export const setTollInfo = (payload) => ({
  type: SET_TOLL_INFO, payload,
})

export const fetchTollInfo = () => ({
  type: FETCH_TOLL_INFO,
})