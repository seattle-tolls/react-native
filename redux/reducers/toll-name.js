import { SWITCH_TOLL_NAME } from '../constants/action-types'
// import { createSelector } from 'reselect'

const initialState = ''

const tollNameReducer = (tollName = initialState, action) => {
  switch (action.type) {
    case SWITCH_TOLL_NAME:
      return action.payload
    default:
      return tollName
  }
}

// TODO: finish the get name selector
export const getName = state => {
  console.log('TOLLNAME-->', state.tollName)
}

export default tollNameReducer