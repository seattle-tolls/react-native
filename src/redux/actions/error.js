import { REPORT_ERROR } from '../constants/action-types'

export const reportError = (error) => ({
  type: REPORT_ERROR, error,
})