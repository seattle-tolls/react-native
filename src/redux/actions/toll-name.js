import { SET_TOLL_NAME } from '../constants/action-types'

export const setTollName = (payload) => ({
  type: SET_TOLL_NAME, payload,
})
