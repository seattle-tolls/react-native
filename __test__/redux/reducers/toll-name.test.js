import tollNameReducer from '../../../src/redux/reducers/toll-name'
import { setTollName } from '../../../src/redux/actions/toll-name'

describe('REDUCERS', () => {
  describe('reducer/toll-name', () => {
    const tollNames = ['toll99', 'toll520']
    const initialState = tollNameReducer('undefined', { type: 'INIT' })
    tollNames.forEach(name => {
      it(`should set the toll name reducer to ${name}`, () => {
        const action = setTollName(name)
        const reducer = tollNameReducer(initialState, action)
        expect(reducer).toBe(name)
        expect(reducer).toMatchSnapshot()
      })
    })

    it('should test for unknown action for tollNameReducer', () => {
      const reducer = tollNameReducer(initialState, { type: 'FAKE_ACTION' })
      expect(reducer).toBe(initialState)
      expect(reducer).toMatchSnapshot()
    })
  })
})