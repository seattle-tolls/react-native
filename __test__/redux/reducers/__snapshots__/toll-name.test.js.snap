// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REDUCERS reducer/toll-name should set the toll name reducer to toll99 1`] = `"toll99"`;

exports[`REDUCERS reducer/toll-name should set the toll name reducer to toll520 1`] = `"toll520"`;

exports[`REDUCERS reducer/toll-name should test for unknown action for tollNameReducer 1`] = `"undefined"`;
